users/vieting/experiments/switchboard/ctc/feat/network_helpers/perturbation.py

```diff
@@ -58,60 +54,51 @@ def __init__(
         """
         self._speed = PerturbationFactor(**speed) if speed else None
         self._tempo = PerturbationFactor(**tempo) if tempo else None
-        self._perturbations = [functools.partial(self.sox, sox_effects=sox_effects)]
+        self._sox_effects = sox_effects or []
         if preemphasis:
             self._perturbations.append(functools.partial(self.preemphasis, factor=PerturbationFactor(**preemphasis)))
-        if codecs:
-            self._perturbations.append(functools.partial(self.apply_codecs, codecs=codecs))
 
     def run(self, audio, sample_rate, random_state):
-        audio = torch.from_numpy(audio).unsqueeze(0).to(torch.float32)
-        for perturbation in self._perturbations:
-            audio = perturbation(audio, sample_rate, random_state)
-        audio = audio.numpy().squeeze()
+        import numpy as np
         assert isinstance(audio, np.ndarray)
         assert len(audio.shape) == 1
+        audio = audio.astype(np.float32)
+        audio = self.sox(audio, sample_rate, random_state)
         return audio
 
-    def sox(self, audio, sample_rate, random_state, sox_effects):
-        sox_effects = sox_effects or []
+    def sox(self, audio, sample_rate, random_state):
+        import random
+        import sox
         speed = False
-        if self._speed is not None:
-            if random_state.random() < self._speed.prob:
-                factor = random_state.random() * (self._speed.max - self._speed.min) + self._speed.min
-                sox_effects.append(["speed", str(factor)])
-                sox_effects.append(["rate", str(sample_rate)])
-                speed = True
-        if self._tempo is not None:
-            if random_state.random() < self._tempo.prob and not speed:
-                factor = random_state.random() * (self._tempo.max - self._tempo.min) + self._tempo.min
-                sox_effects.append(["tempo", str(factor)])
-        audio, _ = torchaudio.sox_effects.apply_effects_tensor(audio, sample_rate, sox_effects)
+        tfm = sox.Transformer()
+        if self._speed is not None and random.random() < self._speed.prob:
+            factor = random.random() * (self._speed.max - self._speed.min) + self._speed.min
+            tfm.speed(factor)
+            speed = True
+        if self._tempo is not None and random.random() < self._tempo.prob and not speed:
+            factor = random.random() * (self._tempo.max - self._tempo.min) + self._tempo.min
+            #tfm.tempo(factor) is more efficiant for larger factors(>1.1).
```

Comment:
I think we should have factors between 0.9 and 1.1 more often than factors outside that range. You can research whether `tfm.tempo` still makes sense, but the comment should be removed or help more clearly. right now it is unclear what should be used.

Reply:
Removed Comment. should be optimal in most cases. The difference is not worth checking if we are outside the range